packages/fxa-payments-server/src/routes/Product/PayPalButton/index.tsx

```diff
@@ -1,19 +1,89 @@
-import React from 'react';
+import React, { useCallback } from 'react';
 import ReactDOM from 'react-dom';
 
+import * as apiClient from '../../../lib/apiClient';
+import { Customer } from '../../../store/types';
+import { SubscriptionCreateAuthServerAPIs } from '../SubscriptionCreate';
+
 declare var paypal: {
   Buttons: {
     driver: Function;
   };
 };
 
-const PaypalButtonBase = paypal.Buttons.driver('react', {
-  React,
-  ReactDOM,
-});
+export type PaypalButtonProps = {
+  apiClientOverrides?: Partial<SubscriptionCreateAuthServerAPIs>;
+  customer: Customer | null;
+  setPaymentError: Function;
+  idempotencyKey: string;
+  ButtonBase?: React.ElementType;
+};
+
+export type ButtonBaseProps = {
+  createOrder?: (
+    event: React.MouseEvent<HTMLButtonElement, globalThis.MouseEvent>
+  ) => void;
+  onError?: (
+    event: React.MouseEvent<HTMLButtonElement, globalThis.MouseEvent>
+  ) => void;
+};
+
+export const PaypalButtonBase =
+  typeof paypal !== 'undefined'
+    ? paypal.Buttons.driver('react', {
+        React,
+        ReactDOM,
+      })
+    : null;
```

Comment:
Hmm what happens if `PaypalButtonBase` is null and no `ButtonBase` is passed in?

Reply:
We get a General Application Error modal currently. I guess an example of this would be if we fail to load the PayPal page script that renders the button?

I could conditionally render the `<ButtonBase>` like so:
```jsx
  return (
    <>
    {ButtonBase && (
      <ButtonBase
        data-testid="paypal-button"
        createOrder={createOrder}
        onError={onError}
      />
    )}
    </>
  );
```
It looks like I'd have to wrap it in a React Fragment. Would we want to log this error e.g. in Sentry? I guess we don't post `fxa-payments-server` errors to our Sentry Slack integrations anymore.

Reply:
> I guess we don't post fxa-payments-server errors to our Sentry Slack integration anymore.

Not to Slack, but they are still reported to Sentry.